public/app/plugins/datasource/prometheus/components/PrometheusMetricsBrowser.tsx

```diff
@@ -370,11 +370,16 @@ export class UnthemedPrometheusMetricsBrowser extends React.Component<BrowserPro
     }
   };
 
-  async fetchValues(name: string) {
+  async fetchValues(name: string, selector: string) {
     const { languageProvider } = this.props;
     this.updateLabelState(name, { loading: true }, `Fetching values for ${name}`);
     try {
       let rawValues = await languageProvider.getLabelValues(name);
+      // If selector changed, clear loading state and discard result by returning early
+      if (selector !== buildSelector(this.state.labels)) {
+        this.updateLabelState(name, { loading: false }, '');
```

Comment:
any reason to explicitly pass an empty string to the `status` param if in `updateLabelState` defaults to the same value?

```suggestion
        this.updateLabelState(name, { loading: false });
```